src/app/info/WikiItemEdit.tsx

```diff
@@ -0,0 +1,138 @@
+"use client";
+
+import { DbWikiRow } from "@/databaseUtils";
+import { WikiEditModeButton, WikiItemAccordionSurface } from "@/app/info/StyleComponents";
+import { TextField } from "@mui/material";
+import DeleteIcon from "@mui/icons-material/Delete";
+import CancelIcon from "@mui/icons-material/Cancel";
+import SaveAltIcon from "@mui/icons-material/SaveAlt";
+import supabase from "@/supabaseClient";
+import { logErrorReturnLogId } from "@/logger/logger";
+import { WikiRowQueryType } from "@/app/info/AddWikiItemButton";
+
+interface WikiItemEditProps {
+    rowData: DbWikiRow;
+    setRowData: (row?: DbWikiRow) => void;
+    setIsInEditMode: (isInEditMode: boolean) => void;
+    sortedRows: DbWikiRow[];
+}
```

Comment:
Two questions here: Firstly, does the edit component really need the entire array of `sortedRows` as a property? And secondly, can `setIsInEditMode` be replaced by `closeEditMode` since we only ever call it with false here